modules/whoisquery/whoisquery.go

```diff
@@ -0,0 +1,96 @@
+package whoisquery
+
+import (
+	"errors"
+	"time"
+
+	"github.com/netdata/go-orchestrator/module"
+	"github.com/netdata/go.d.plugin/pkg/web"
+)
+
+func init() {
+	creator := module.Creator{
+		Defaults: module.Defaults{
+			UpdateEvery: 60,
+		},
+		Create: func() module.Module { return New() },
+	}
+
+	module.Register("whoisquery", creator)
+}
+
+func New() *WhoisQuery {
+	return &WhoisQuery{
+		Config: Config{
+			Timeout:       web.Duration{Duration: time.Second * 2},
+			DaysUntilWarn: 14,
+			DaysUntilCrit: 7,
+		},
+	}
+}
+
+type Config struct {
+	web.ClientTLSConfig `yaml:",inline"`
+	Timeout             web.Duration
```

Comment:
Timeout and web.ClientTLSConfig are not used

the library you are using has hardcoded timeouts

https://github.com/likexian/whois-go/blob/92cfa71d7edfc954b9661b2676edfb360c32eb3c/whois.go#L102